remme/tp/pub_key.py

```diff
@@ -318,6 +330,91 @@ def _store_pub_key(self, context, signer_pubkey, transaction_payload):
+
+        transaction_payload (pub_key_pb2.NewPubKeyStoreAndPayPayload): ...
+        """
+        new_public_key_payload = transaction_payload.pub_key_payload
+
+        secp256k1_public_key = Secp256k1PublicKey.from_hex(transaction_payload.owner_public_key.decode())
+
+        is_owner_public_key_payload_signature_valid = Secp256k1Context().verify(
+            signature=transaction_payload.signature_by_owner.decode(),
+            message=new_public_key_payload.SerializeToString(),
+            public_key=secp256k1_public_key,
+        )
+        if not is_owner_public_key_payload_signature_valid:
+            raise InvalidTransaction('Public key owner\'s signature is invalid!')
+
+        conf_name = new_public_key_payload.WhichOneof('configuration')
+        if not conf_name:
+            raise InvalidTransaction('Configuration for public key not set')
+
+        conf_payload = getattr(new_public_key_payload, conf_name)
+
+        processor_cls = detect_processor_cls(conf_payload)
+        processor = processor_cls(new_public_key_payload.entity_hash,
+                                  new_public_key_payload.entity_hash_signature,
+                                  new_public_key_payload.valid_from,
+                                  new_public_key_payload.valid_to,
+                                  new_public_key_payload.hashing_algorithm,
+                                  conf_payload)
+
+        sig_is_valid = processor.verify()
+        if sig_is_valid is False:
+            raise InvalidTransaction('Payed public key has invalid signature.')
+
+        public_key = processor.get_public_key()
+        public_key_to_store_address = self.make_address_from_data(public_key)
+
+        public_key_to_store_owner_address = AccountHandler().make_address_from_data(transaction_payload.owner_public_key)
+        payer_for_storing_address = AccountHandler().make_address_from_data(signer_pubkey)
+
+        public_key_information, public_key_to_store_owner_account, payer_for_storing_account = get_multiple_data(context, [
+            (public_key_to_store_address, PubKeyStorage),
+            (public_key_to_store_owner_address, Account),
+            (payer_for_storing_address, Account),
+        ])
+
+        if public_key_information:
+            raise InvalidTransaction('This public key is already registered.')
+
+        if payer_for_storing_account is None:
+            payer_for_storing_account = Account()
+
+        if not self._is_public_key_validity_exceeded(
+            valid_from=new_public_key_payload.valid_from,
+            valid_to=new_public_key_payload.valid_to,
+        ):
+            raise InvalidTransaction('The public key validity exceeds the maximum value.')
+
+        public_key_information = PubKeyStorage()
+        public_key_information.owner = public_key_to_store_owner_address
+        public_key_information.payload.CopyFrom(new_public_key_payload)
+        public_key_information.is_revoked = False
+
+        state = {
```

Comment:
You forgot to update the owner account (see `_store_pub_key`).

Reply:
@eugene-babichenko, done.